lib/winston/container.js

```diff
@@ -55,16 +55,43 @@ module.exports = class Container {
   }
 
   /**
-   * Retreives a `winston.Logger` instance for the specified `id`. If
-   * an instance does not exist, one is created.
-   * @param {!string} id - The id of the Logger to get.
-   * @param {?Object} [options] - Options for the Logger instance.
+ * @deprecated Use `loggers.getOrCreate(id, options)` or `loggers.add(id, options)`.
+ *
+ * Retrieves a `winston.Logger` instance for the specified `id`.
+ * If an instance does not exist, one is created.
+ * ⚠️ Passing `options` here will only be effective when creating,
+ * not when retrieving.
+ *
+ * @param {!string} id - The id of the Logger to get.
+ * @param {?Object} [options] - Options for the Logger instance.
+ * @returns {Logger} - A configured Logger instance with a specified id.
+ */
+  get(id, options) {
+    process.emitWarning(
+      'loggers.get(id, options) is deprecated — use loggers.getOrCreate(id, options) or loggers.add(id, options).',
+      'DeprecationWarning'
+    );
+    return this.getOrCreate(id, options);
+  }
+
+  /**
+   * Retrieves a `winston.Logger` instance for the specified `id`,
+   * or creates it with the given options if it does not exist.
+   *
+   * @param {!string} id - The id of the Logger to get or create.
+   * @param {?Object} [options] - Options for the Logger instance
+   *   (only used if creating).
    * @returns {Logger} - A configured Logger instance with a specified id.
    */
-  get(id, options) {
-    return this.add(id, options);
+  getOrCreate(id, options) {
+    if (this.has(id)) {
```

Comment:
Where is this `has` function defined? I am only aware of `this.loggers.has`, which is attached to `this.loggers = new Map()`

Reply:
Ah I have found this one 😄  odd that it didnt appear in my code search, you can safely disregard this comment.